packages/flutter/lib/src/rendering/mouse_cursor.dart

```diff
@@ -0,0 +1,270 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'mouse_tracking.dart';
+
+/// An interface for mouse cursor definitions.
+///
+/// A mouse cursor is a graphical image on the screen that echoes the movement
+/// of a pointer device, such as a mouse or a stylus. A [MouseCursor] object
+/// defines a kind of mouse cursor, such as an arrow, a pointing hand, or an
+/// I-beam.
+///
+/// Internally, when the mouse pointer moves, it finds the front-most region
```

Comment:
In some way this is intentionally put in a vague way, because it depends on different layers of Flutter. It can be an `AnnotatedRegionLayer<MouseTrackerAnnotation>`, `RenderMouseRegion`, or `MouseRegion`. There are also other render objects or widgets that do not directly use `(Render)MouseRegion` but directly push annotations, such as `PlatformView`, among a few.

The only thing we're certain of is that cursors have to be assigned to logical regions of the screen.